### PR TITLE
Add partitions to allow running a specific subset of stories

### PR DESCRIPTION
**Pull request by ronakj**

### CODE REVIEW

File: src/StoryWrightProcessor/StoryWrightProcessor.ts [hunk 1]
```diff
@@ -27,9 +27,20 @@ export class StoryWrightProcessor {
...
+          console.log(
+            "Starting partitioning with", options.totalPartitions, "total partitions and partition number to select", options.partition
+          );
+          console.log("Total Stories length", stories.length);
+          const totalStories = stories.length;
+          const storiesInEachPartition = Math.ceil(totalStories/options.totalPartitions);
+          const startingIndex = (options.partition - 1)  * storiesInEachPartition;
+          console.log("Partition indexes", startingIndex, startingIndex + storiesInEachPartition);
+          stories = stories.slice(startingIndex, startingIndex + storiesInEachPartition);
+        }
```
Comment (sunilsurana) [inline src/StoryWrightProcessor/StoryWrightProcessor.ts:43]: Move all this in a utils to keep main logic clean stories=Utils.createPartitions(options)

**Comment by p01** [comment]
The term "partition" rings of file system partitions. The terms "slice" and "slices" might be more generic.

File: src/main.ts [hunk 2]
```diff
@@ -37,6 +37,16 @@ const args = argv.usage('Usage: $0 [options]').help('h').alias('h', 'help')
...
         type: 'boolean'
     })
+    .option('partition', {
+        default: 1,
+        describe: "Partition number to run, used in conjunction with totalPartitions parameter to run specific partition of the stories",
+        type: "number"
+    })
+    .option('totalPartitions', {
+        default: 1,
+        describe: "Number of partitions, used in conjunction with partition parameter to run specific partition of the stories",
```
Comment (p01) [inline src/main.ts:47]: To be honest, I took me a few seconds to get the purpose of the partitions. We should explicitly mention in their description that the purpose to enable to distribute the work across multiple tasks/agents/threads/workers.

File: src/StoryWrightProcessor/StoryWrightProcessor.ts (hunk 1 above)
Comment (ronakj) [reply src/StoryWrightProcessor/StoryWrightProcessor.ts:42]: Updated.

File: src/main.ts (hunk 2 above)
Comment (ronakj) [reply src/main.ts]: Updated.

**Review by sunilsurana — APPROVED** [review]

**Comment by ronakj** [comment]
> The term "partition" rings of file system partitions. The terms "slice" and "slices" might be more generic.

I think that might also be confusing. "slice" in javascript is typically used in array.slice where it's expected to get the starting index and ending index. 
If we use slice then the parameter for slice/sliceIndex may be confusing. @p01

**Review by p01 — APPROVED** [review]

**Pull request merged**